tests/unit/datastore_test.py

```diff
@@ -90,6 +331,586 @@ def test_profile_update_rules(self):
+        expected_calls = [call(TEST_PROFILE_PATH + "tags", '["TEST"]'),
+                          call(TEST_PROFILE_PATH + "rules", rules.to_json())]
+        self.etcd_client.write.assert_has_calls(expected_calls, any_order=True)
+
+    def test_delete_profile(self):
+        """
+        Test deleting a policy profile.
+        """
+        self.datastore.delete_profile("TEST")
+        self.etcd_client.delete.assert_called_once_with(TEST_PROFILE_PATH,
+                                                        recursive=True,
+                                                        dir=True)
+
+    def test_get_profile_names_2(self):
+        """
+        Test get_profile_names() when there are two profiles.
+        """
+        self.etcd_client.read.side_effect = mock_read_2_profiles
+        profiles = self.datastore.get_profile_names()
+        self.assertSetEqual(profiles, {"UNIT", "TEST"})
+
+    def test_get_profile_names_no_key(self):
+        """
+        Test get_profile_names() when the key hasn't been set up.  Should
+        return empty set and not raise a KeyError.
+        """
+        self.etcd_client.read.side_effect = mock_read_profiles_key_error
+        profiles = self.datastore.get_profile_names()
+        self.assertSetEqual(profiles, set())
+
+    def test_get_profile_names_no_profiles(self):
+        """
+        Test get_profile_names() when there are no profiles.
+        """
+        self.etcd_client.read.side_effect = mock_read_no_profiles
+        profiles = self.datastore.get_profile_names()
+        self.assertSetEqual(profiles, set())
+
+    def test_get_profile_members(self):
+        """
+        Test get_profile_members() when there are endpoints.
+        """
+        self.etcd_client.read.side_effect = mock_read_4_endpoints
+        members = self.datastore.get_profile_members("TEST")
+        self.assertSetEqual({"567890abcdef", "7890abcdef12"},
+                            set(members))
+
+        members = self.datastore.get_profile_members("UNIT")
+        self.assertSetEqual({"90abcdef1234", "1234567890ab"},
+                            set(members))
+
+        members = self.datastore.get_profile_members("UNIT_TEST")
+        self.assertSetEqual(set(), set(members))
+
+    def test_get_profile_members_no_key(self):
+        """
+        Test get_profile_members() when the endpoints path has not been set up.
+        """
+        self.etcd_client.read.side_effect = mock_read_endpoints_key_error
+        members = self.datastore.get_profile_members("UNIT_TEST")
+        self.assertSetEqual(set(), set(members))
+
+    def test_get_endpoint_exists(self):
+        """
+        Test get_endpoint() for an endpoint that exists.
+        """
+        ep = Endpoint("1234567890ab", "active", "11-22-33-44-55-66")
+        self.etcd_client.read.side_effect = mock_read_for_endpoint(ep)
+        ep2 = self.datastore.get_endpoint("TEST_HOST", "1234", "1234567890ab")
```

Comment:
I'd prefer to use explicit kwargs here, since it's not immediately obvious what "1234" is.

Making the change above might obviate the need to be more explicit though.